syncmap/syncmap.go

```diff
@@ -0,0 +1,62 @@
+package syncmap
+
+import (
+	"iter"
+	"sync"
+)
+
+// Map is a regular map but synchronized with a mutex.
+type Map[K comparable, V any] struct {
+	mu sync.Mutex
+	m  map[K]V
+}
+
+// New returns a new syncmap.
+func New[K comparable, V any]() *Map[K, V] {
+	return &Map[K, V]{
+		m: make(map[K]V),
+	}
+}
+
+// Load returns the value for a key.
+func (m *Map[K, V]) Load(key K) (V, bool) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	v, ok := m.m[key]
+	return v, ok
+}
+
+// Store sets the value for a key.
+func (m *Map[K, V]) Store(key K, value V) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.m[key] = value
+}
+
+// Delete deletes a key.
+func (m *Map[K, V]) Delete(key K) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	delete(m.m, key)
+}
+
+// Len returns the number of elements in the map.
+func (m *Map[K, V]) Len() int {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return len(m.m)
+}
+
+// Range calls f sequentially for each key and value present in the map.
+// If f returns false, range stops the iteration.
+func (m *Map[K, V]) All() iter.Seq2[K, V] {
```

Comment:
```go
m.mu.Lock()
defer m.mu.Unlock()
return maps.All(maps.Clone(m.m))

Reply:
did something else that should still be correct. I've tried using testing/quick to test it but I don't think my usage of it is conclusive.